target/scripts/startup/setup.d/log.sh

```diff
@@ -13,31 +13,25 @@ function _setup_logs_general() {
 function _setup_logrotate() {
   _log 'debug' 'Setting up logrotate'
 
-  LOGROTATE='/var/log/mail/mail.log\n{\n  compress\n  copytruncate\n  delaycompress\n'
-
-  case "${LOGROTATE_INTERVAL}" in
-    ( 'daily' )
-      _log 'trace' 'Setting postfix logrotate interval to daily'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  daily\n"
-      ;;
-
-    ( 'weekly' )
-      _log 'trace' 'Setting postfix logrotate interval to weekly'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  weekly\n"
-      ;;
-
-    ( 'monthly' )
-      _log 'trace' 'Setting postfix logrotate interval to monthly'
-      LOGROTATE="${LOGROTATE}  rotate 4\n  monthly\n"
-      ;;
-
-    ( * )
-      _log 'warn' 'LOGROTATE_INTERVAL not found in _setup_logrotate'
-      ;;
-
-  esac
-
-  echo -e "${LOGROTATE}}" >/etc/logrotate.d/maillog
+  if [[ ${LOGROTATE_INTERVAL} =~ ^(dai|week|month)ly$ ]]; then
+    _log 'trace' "Logrotate interval set to ${LOGROTATE_INTERVAL}"
+  else
+    _log 'warn' "Value '${LOGROTATE_INTERVAL}' for LOGROTATE_INTERVAL is invalid - falling back to default"
+    export LOGROTATE_INTERVAL='weekly'
+    # shellcheck disable=SC2034
+    VARS[LOGROTATE_INTERVAL]='weekly'
+  fi
+
+  cat >/etc/logrotate.d/maillog << EOF
+/var/log/mail/mail.log
+{
+  compress
+  copytruncate
+  delaycompress
+  rotate 4
+  ${LOGROTATE_INTERVAL}
+}
+EOF
```

Comment:
**No action required**

This is much more readable than the previous approach with the split `{` + `}` that looked almost like a typo 🚀 

Regarding [`rotate 4`](https://linux.die.net/man/8/logrotate), I guess that makes sense for `weekly`, while a bit odd for `daily` and `monthly`. I'm assuming `git blame` would show this was introduced with `weekly` in mind.

Given you've dropped the switch statement and no one seems to be bothered I guess leaving it at `4` here is fine, I was just a little curious what that setting did when I saw it here.

Reply:
I stumbled upon the same while reviewing. We could introduce `$LOGROTATE_COUNT` or (not so readable) `$LOGROTATE_ROTATE` to make the rotation count configurable.

Reply:
It should be ok, no need to add new ENV to support this until we get actual requests to adjust the default.